### ISSUE TITLE
Shortucts in napari require click on canvas to switch ViewerModel in context.

### ISSUE THREAD

**Issue by Czaki** [napari/napari#4726]
## 🐛 Bug

Move mouse above additional canvas (QtWiewer) docked in plugin are does not switch context for a shortcut. Context is switched only when clicking on the canvas. Because vispy does not provide the proper way to inform about leaving canvas then shortcut may use wrong data (like mouse coordinate). 

## To Reproduce

1. switch to #4665
2. start the example from PR
3. load image and enable cross
4. move the mouse above the additional canvas, click the mouse and then click `C` 
5. see that cross is properly moved
6. move mouse above another canvas and click `C` 
7. see that warning that the mouse is not over the canvas is shown. 

## Expected behavior

Move mouse over the canvas in the active window should be enough to switch context. 

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

 - Any other relevant information:

## Additional context

This need to activate vispy canvas may be the source of @DragaDoncila problem that part of the shortcuts is not working when testing #4628.

**Comment by DragaDoncila** [comment]
@Czaki I think this was fixed by #4768?

**Comment by Czaki** [comment]
No. There is a try to fix this in #4762, but it has too many side effects.
